python/thunder/rdds/series.py

```diff
@@ -56,6 +56,22 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise ValueError("Index must be a non-empty indexable object")
+        if self._index is None:
+            self.populateParametersFromFirstRecord()
+        try:
+            lenMatch = len(value) == len(self._index)
+        except:
+            raise ValueError("Index must be an object with a length")
+        if not lenMatch:
+            raise ValueError("New index length must match that of the current index")
```

Comment:
I can imagine somebody getting really frustrated here if they think the length of the new index they're passing *does* already match the length of the current index. How about adding printouts of the length of the new index and the existing index to the exception text? something like `"New index length (%d) ...current index (%d)" % (len(value), len(self._index))`

Also I don't think the `lenMatch` var is really needed here, Could do:
```python
try:
  if len(val) != len(self._index):
    raise blah blah
except:
  raise another blah
```